proj/urls.py

```diff
@@ -2,19 +2,20 @@
 from django.urls import path, include
 from django.conf.urls.static import static
 from django.conf import settings
-from products.views import products, index
+from products.views import products, index, single
 
 
 apipatterns = [
     path('', include('products.urls')),
-    path('products/<int:product_id>/comments', include('comments.urls'))
+    path('products/<int:product_id>/', include('comments.urls'))
 ]
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('api/', include((apipatterns, 'api'), namespace='api')),
     path('api-auth/', include('rest_framework.urls', namespace='rest_framework')),
     path('', index, name='index'),
-    path('products/', products, name='products')
+    path('products/', products, name='products'),
+    path('single/', single, name='single'),
```

Comment:
How are you going to understand which product page a user wants to open?

Reply:
replace this row to
`path('products/<int:product_id>/', single, name='single')`

and get pass `product_id` to the template